### DISCUSSION TITLE
Welcome to pico-WSPRer Discussions!

### DISCUSSION THREAD

**Discussion by EngineerGuy314** [EngineerGuy314/pico-WSPRer#30] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by serych** [comment]
Hi Rob!
Thanks for creating the discussions here!
My first topic is about **solution of different boards for pico WSPRer**. I have seen, that you created your own board, also I have my board https://github.com/serych/pico-HABakuk (just now in production in JLCPCB) and there are other pico clones with smaller to bigger hardware differences. It mainly concerns the LED used (classical or WS 1228 type) and the pin it is connected to, but it can also be other changes (for example on my board I need to invert the signal to switch the GSP module because I have a P type MOSFET on the VCC side (I want to keep GPS backup power on to keep GPS almanac).
**There are basically two ways to deal with this.** The first is the one you chose when solving your board - add a **menu item and select the board at runtime**. The second way is to **create a file with a set of #define and #ifdef sections** and define the differences of the different boards there. 
The only advantage of the first way is that you can publish a compiled u2f file on github and everyone can use it right away. 
On the other hand, I think the second way is much clearer, other people can easily add their boards as needed, but it has the disadvantage that the project has to be recompiled after changing the board. 
What do you think? Which way to go? 
Jakub, OK1SE

**Reply by EngineerGuy314** [reply]
oh wow, you also made a custom board! I didn't realize so many people were doing that! 

This was my very first time doing any kind of PCB design, it was a difficult learning experience to say the least, and my result is definitely not as refined as yours. But I am very shocked at how cheap and easy it is. 

Custom PCBs are nice, but I want to try to keep the program compatible with the original method of gluing a GPS module to a Pico. I was going to make a separate fork, but I think user-configurable IO mappings are a better solution.  Easier than synchronizing future updates/enhancements across forks or branches.

I also definitely want to avoid having to recompile to incorporate changes. It seems nearly half of the people that have expressed in this project don't have the ability or comfort level with setting up the SDK and recompiling the program.

But with all the TELEN stuff I plan to add, the existing user-config menu is becoming cumbersome.  I think the menu code needs a major re-think and re-write, or some other method is needed. I was thinking maybe having all the config options in a text file that is uploaded to the board? Zachtek boards use a separate .exe  you run on your PC for configuration, and Traquito uses a web based config tool. But I like the elegance of doing everything with a simple terminal. I'm open to ideas.

**Comment by tomplant** [comment]
Howdy! 
I'm Tom, and I stumbled upon this while looking about for a dirt-cheap WSPR beacon. I figured I'd test things out before taking a "deeper dive" into looking into doing anything Balloon-Launchy, mostly before expanding the project(s) to a higher cost as right now I managed to get 20 Pico's at $2.75usd/ea since I buy things in bulk a lot.

Unfortunately it seems I've not been able to get things to work well- at all. Even though I have no GPS chip attached, I was under the impression that forcing the tuning mode would cause the program to transmit and having connected it to a 20m 1/2 dipole in the shack, I detect no RF output what-so-ever. I know you've received the suggestion to update the coding/config for use as a stationary tracker. I'd like to echo that suggestion, as I'd love to see this as a viable dirt-cheap WSPR beacon for general use.

I figure once I can "proof of concept" things to my own brain, I'll look into getting some boards made, etc and perhaps let a balloon go sailing. Plus I think with a little bit of prodding I could get not only my club interested, some teachers in the local school district, but the local CBS affiliate interested (more than 1 station member is a Ham, and we even have one of the reporters in our pocket so to speak :-) (They actually did a really nice spot on Amateur Radio a couple of times already).  

Cheers!
Tom KB2URI
VP Niagara Frontier Radiosport (NF2RS)

**Reply by EngineerGuy314** [reply]
>> for use as a stationary tracker. I'd like to echo that suggestion, as I'd love to see this as a viable dirt-cheap WSPR beacon for general use.

Once I get the user-config menu revamped, that will definitely be added as an option

>> I was under the impression that forcing the tuning mode would cause the program to transmit

i've never actually used the tuning mode (it was implemented by serych), but that is how its supposed to work. It won't actually transmit data though, just a carrier.

did you check the output with a scope? If there isn't a GPS module, the output frequency will be uncalibrated, and probably way off.

**Reply by EngineerGuy314** [reply]
I tried the tune mode, and for some reason it was truncating the value (asked for 14.5Mhz, but it put out 14Mhz instead). I pushed a change just now that fixed it

**Reply by EngineerGuy314** [reply]
>> I know you've received the suggestion to update the coding/config for use as a stationary tracker.

For now I made a change so that if you disable BOTH the Zachtek and U4B protocols (by entering - for suffix and -- for id13) it will do standard beacon mode (a single Type 1 message per 10 minute period, no suffix on callsign, 4 char Grid only, and hardcoded 10 for power). Very basic operation only, no frequency hopping, no band hopping. ~~Testing it now, will push the update later.~~ done

**Reply by nilspupils** [reply]
Thanks for implementing that!
Nils

**Comment by alanmimms** [comment]
I'm working on a WSPR transmitter design for 80m-10m with ESP32 WiFi based UI for scheduling, control, etc. and Si5351 and GNSS based 1pps signal continuous calibration. I want to call my project "WSPRer", but I see you already used that name. I wonder if I can use it too - perhaps with something else added before or after to avoid confusion? I'm considering selling these things via Crowdsupply or similar after the design is vetted, so it matters a little. I'm not planning to do any trademark or similar. The project is entirely MIT license open source for hardware and software. It's a work-in-progress at this point, but it's fun and I think I might manage to finish it by end of this year or early 2027.

Thanks,
Alan WB7NAB

**Comment by EngineerGuy314** [comment]
Hello Alan,
I see no problem with calling your project WSPRer, especially if you add a prefix or suffix of some sort.
ESP32's are great, i've used them for a few other projects. Amazing how something so cheap can be so powerful...
Best of luck to you.RobKC3LBR
    On Wednesday, July 8, 2026 at 02:25:02 PM EDT, Alan Mimms ***@***.***> wrote:  
 
 


I'm working on a WSPR transmitter design for 80m-10m with ESP32 WiFi based UI for scheduling, control, etc. and Si5351 and GNSS based 1pps signal continuous calibration. I want to call my project "WSPRer", but I see you already used that name. I wonder if I can use it too - perhaps with something else added before or after to avoid confusion? I'm considering selling these things via Crowdsupply or similar after the design is vetted, so it matters a little. I'm not planning to do any trademark or similar. The project is entirely MIT license open source for hardware and software. It's a work-in-progress at this point, but it's fun and I think I might manage to finish it by end of this year or early 2027.

Thanks,
Alan WB7NAB

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by alanmimms** [comment]
Thanks. I'm thinking "HF WSPRer". I'm considering augmenting the hardware
to make it possible to use it as QRP up to 10W CW and even SSB transmitter
my adding a tracking buck/boost converter to provide power control and
amplitude modulation while i control phase using my CPLD. This converter is
only necessary if i decide to modulate, but the idea came when i was
considering how to control tx power with some accuracy. 73s! Alan WB7NAB.
